File: JAVA/Insertion_Sort.java
import java.util.*;
public class Insertion_Sort{

    public static void printArr(int arr[]){
        System.out.println("Sorted array: ");
        for(int i=0; i<arr.length; i++){
            System.out.print(arr[i]+" ");
        }    
    }
    public static void main(String args[]){
        Scanner sc= new Scanner(System.in);

        System.out.print("Enter the number of elements: ");
        int n= sc.nextInt();

        int arr[]=new int[n];

        System.out.println("Enter the elements: ");
        for(int i=0; i<n; i++){
            arr[i]=sc.nextInt();
        }

        //  Insertion Sort
        for (int i=1; i<arr.length; i++){
            int current= arr[i];
            int j=i-1;

            while(j>=0 && arr[j]>current){
                arr[j+1]=arr[j];
                j--;
            }
            arr[j+1]=current;
        }
        printArr(arr);
    }
}

File: C++/InsertionSort.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){

    int n, temp;
    cout<<"Enter the size of array: ";
    cin>>n;

    int arr[n];
    cout<<"Enter the elements of array: ";
    for(int i=0;i<n;i++){
        cin>>arr[i];
    }

    for(int i=1; i<n; i++){     // i is tracking unsorted array
        int current= arr[i];    //current is the element to be placed at its correct position
        int j=i-1;              //j is tracking sorted array
        
        while(j>=0 && current< arr[j]){
            arr[j+1]=arr[j];
            j--;
        }
        //placement of current element
        arr[j+1]=current;
    }

    cout<<"Sorted array is: ";
    for(int i=0;i<n;i++){
        cout<<arr[i]<<" ";
    }
}